Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp

```diff
@@ -971,6 +971,34 @@ bool LineLayout::hitTest(const HitTestRequest& request, HitTestResult& result, c
     return false;
 }
 
+void LineLayout::shiftLinesBy(LayoutUnit blockShift)
+{
+    if (!m_inlineContent)
+        return;
+    bool isHorizontalWritingMode = WebCore::isHorizontalWritingMode(flow().style().writingMode());
+
+    for (auto& line : m_inlineContent->displayContent().lines)
+        line.moveInBlockDirection(blockShift, isHorizontalWritingMode);
+
+    for (auto& box : m_inlineContent->displayContent().boxes) {
+        if (isHorizontalWritingMode)
+            box.moveVertically(blockShift);
+        else
+            box.moveHorizontally(blockShift);
+    }
+
+    for (auto& object : m_boxTree.renderers()) {
+        Layout::Box& layoutBox = *object->layoutBox();
+        if (layoutBox.isOutOfFlowPositioned() && layoutBox.style().hasStaticBlockPosition(isHorizontalWritingMode)) {
+            CheckedRef renderer = downcast<RenderBox>(m_boxTree.rendererForLayoutBox(layoutBox));
+            ASSERT(renderer->layer());
+            CheckedRef layer = *renderer->layer();
+            layer->setStaticBlockPosition(layer->staticBlockPosition() + blockShift);
+            renderer->setChildNeedsLayout(MarkOnlyThis);
```

Comment:
what guarantees that we end up running layout on this renderer?

Reply:
Nothing inherent to the line layout object, but my logic goes:

- for everything that we're a containing block for, we're running positioned box layout after the align-content shifting
- for everything that our ancestors are a containing block for, they need to wait until we're done with layout to be able to place static positioned boxes that are our descendants anyway, so they need to be running positioned box layout after we're done here anyway

It's not foolproof logic, and there's possibly something I missed here, but I think that should capture most of it...